linode/instancereservedipassignment/framework_resource.go

```diff
@@ -0,0 +1,293 @@
+package instancereservedipassignment
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+// For internal only: please refer to KB page for more information
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_reserved_ip_assign",
```

Comment:
Could this resource be renamed to `linode_reserved_ip_assignment` to maintain declarative naming?

Reply:
I think it makes sense to keep the name "linode_reserves_ip_assign" since the endpoint also has "assign" in it. I could change the name of the folder from "instancereservedipassignment" to "instancereservedipassign"

Reply:
On second thoughts, I agree that linode_reserved_ip_assignment makes more sense for our application. I will making that change and pushing it now